src/jit/importer.cpp

```diff
@@ -18277,7 +18277,11 @@ void Compiler::impInlineInitVars(InlineInfo* pInlineInfo)
         // the inlining multiplier) for anything in that assembly.
         // But we only need to normalize it if it is a TYP_STRUCT
         // (which we need to do even if we have already set foundSIMDType).
-        if ((!foundSIMDType || (sigType == TYP_STRUCT)) && isSIMDClass(&(lclVarInfo[0].lclVerTypeInfo)))
+        if ((!foundSIMDType || (sigType == TYP_STRUCT)) && (isSIMDClass(&(lclVarInfo[0].lclVerTypeInfo))
```

Comment:
Why can't the new logic in `isHWSIMDClass` just be added to `isSIMDClass`?

Reply:
Because `isSIMDClass` is used for `S.N.Vector` recognition. https://github.com/dotnet/coreclr/blob/master/src/jit/simd.cpp#L137